session/txnmanager.go

```diff
@@ -52,6 +57,22 @@ type txnManager struct {
 
 	// We always reuse the same OptimisticTxnContextProvider in one session to reduce memory allocation cost for every new txn.
 	reservedOptimisticProviders [2]isolation.OptimisticTxnContextProvider
+
+	// used for slow transaction logs
+	events          []event
+	lastInstant     time.Time
+	enterTxnInstant time.Time
```

Comment:
Maybe `lastInstant` is unnecessary, we can record the start instant and the duration (since the start instant) of each event, then the gaps can be inferred from these info.

Reply:
Yeah I've considered the alternative. I think they are basically equivalent and either is fine. If you prefer this style I can change it

Reply:
I think it’s fine. A field of type time.Time here is not expensive at all.